tests/python_package_test/test_engine.py

```diff
@@ -29,6 +30,34 @@ def dummy_obj(preds, train_data):
     return np.ones(preds.shape), np.ones(preds.shape)
 
 
+def mse_obj(y_pred, dtrain):
+    y_true = dtrain.get_label()
+    grad = (y_pred - y_true)
+    hess = np.ones(len(grad))
+    return grad, hess
+
+
+def logloss_obj(preds, train_data):
```

Comment:
to avoid introducing a new import from a dependency (`scipy.special`), could you please just copy over the existing implementation from the Dask tests which uses only `numpy`?

https://github.com/microsoft/LightGBM/blob/d163c2c16629e3da41dc5d7b8dbb76b8fc6249db/tests/python_package_test/test_dask.py#L241-L245

(but modified to take in `(preds, train_data)` instead of `(y_true, y_pred)`)

Reply:
copy that